Merge batch ipc status

fredkingham opened the pull request:
Makes the batch IPC status load work with merged MRNs

Commit:
Changes ipc.loader.load_all_extra_ipc_patients to look at merged mrns

Changes the load_all_extra_ipc_patients to also check the merged MRN
table before creating patients.

Commit:
Changes the IPC episode loader to work with merged MRNs

Previously we assumed we could bridge bed status and infection control
by the value provided in the upstream MRN.

Now that we are accounting for patient merges this is no longer True.

We now get the patients with get_mrn_to_patient which takes into account
merged MRNs.

We then match with bed status using patient_id as we assume bed status
already takes into account merged MRNs.

Commit:
Merge branch 'cerner-merge-branch' into merge-batch-ipc-status-1

Commit:
Renames get_patients_from_mrns to find_patients_from_mrns

davidmiller commented on plugins/ipc/management/commands/load_ipc_status.py:
```diff
@@ -84,17 +85,23 @@ def handle(self, *args, **options):
+            if not patient:
+                continue
 
+            if patient.id in inpatient_ids:
```
We might have done this for some testing purpose, but this (and L88-90) above shouldn't have crept into a release branch - suggest we take it out now in case we forget it's here

fredkingham replied:
Are we talking about either

a) Removing IPC creation
b) Changing it so that regardless of bed status IPC statuses are created.

davidmiller replied:
As far as I can tell, the behaviour here is still not as we want it - am I missing where this script creates the patient?

fredkingham commented:
This loader is only going to be run once. 

Therefore we want to load for everyone and create episodes without constraining it to bed status.

Remove the filtering by bed status.

Commit:
Removes the unused "updated" variable in the ipc status loader

Commit:
Makes the load_ipc_status mgmt command load all patients

Previously it only loaded patients who had a bed status but constraint
was for testing purposes only.

Commit:
Adds a comment to the load ipc bed status command

Adds a comment to the load ipc bed status command to explain why a
patient might not be present in the MRN to patient dictionary

davidmiller commented on plugins/ipc/loader.py:
```diff
@@ -26,12 +26,22 @@ def load_all_extra_ipc_patients():
-        all_tested_mrns.update([r['Patient_Number'] for r in tested_mrns])
-
-    all_mrns = set(
+        # elcid does not use zero prefixed MRNs but the lab test
```
This should probably be signalled harder, but this whole function is more of a development/testing thing, related to the initial development of `ipc.models.InfectionAlert` - that's paused for now, I think we'd just 'do this differently'. 
We should probably delete the whole function and anywhere it's called rather than add complexity

fredkingham replied:
cool, in that case we aren't using anything in plugins.ipc.loader so I have deleted it.

fredkingham commented:
This needs further conversation see https://github.com/orgs/openhealthcare/projects/1/views/1?pane=issue&itemId=22173238

Commit:
Removes ipc.loader as its not used.

ipc.loader contained the function load_all_extra_ipc_patients is only
there for development and tesing
(https://github.com/openhealthcare/elcid-rfh/pull/1373/files#r1115543430)

fredkingham commented:
This still needs to be changed to create patients

Commit:
Changes the IPC Status load to create patients

The IPC status load should create patients if they don't exist
